modo/cram.py

```diff
@@ -10,20 +10,40 @@
 from rdflib import Graph
 import modo_schema.datamodel as model
 
-from .helpers import parse_region
+import os
+from .helpers import parse_region, htsget_command
 
 
 def slice_cram(path: str, region: str) -> Iterator[AlignedSegment]:
     """Return an iterable slice of the CRAM file."""
 
-    chrom, start, stop = parse_region(region)
+    chrom, start, end = parse_region(region)
+    if start == "":
+        start = None
+    else:
+        start = int(start)
+    if end == "":
+        end = None
+    else:
+        end = int(end)
     cramfile = AlignmentFile(path, "rc")
 
-    iter = cramfile.fetch(chrom, start, stop)
+    iter = cramfile.fetch(chrom, start, end)
 
     return iter
 
 
+def slice_remote_cram(
+    url: str, region: str = None, output_filename: str = None
+):
+    """Stream or write to a local file a slice of a remote CRAM file"""
+
+    htsget_req = htsget_command(url, region, output_filename)
+    os.system(htsget_req)
```

Comment:
Yes, now that I can actually use sys.stdout.buffer ... which was not possible via Spyder ... I will change the script to use the htsget api instead of the cli via os.system() .

Reply:
Implemented (see commit - https://github.com/sdsc-ordes/modo-api/pull/31/commits/833573db60b063778c1901f90a055e74fb97bbd0)